docs/GUIDE.md

```diff
@@ -0,0 +1,379 @@
+      <button id="mic-toggle" class="media-active">Mic</button>
+      <button id="video-toggle" class="media-active">Video</button>
+      <button id="leave-channel" class="media-active">Leave</button>
+    </div>
+    <div id="overlay" class="modal">
+      <div id="form-container">
+        <h1 id="form-header">Avatar Video Chat</h1>
+        <form id="join-channel-form">
+          <div id="form-body">
+            <div class="form-group">
+              <label for="form-rpm-url">Ready Player Me URL</label>
+              <input type="text" id="form-rpm-url" placeholder="http://models.readyplayer.me/<MODEL-ID>.glb" class="form-control">
+            </div>
+            <div id="form-footer">
+              <button type="submit" id="join-channel-btn">Join Channel</button>
+            </div>
+          </div>
+        </form>
+      </div>
+    </div>
+    <script type="module" src="/main.js"></script>
+  </body>
+</html>
+```
+
+## Agora Client and data stores
+In [`main.js`](/main.js) we create a new Agora client to use Agora's SDK and use `localMedia` to keep a reference to the audio, video, and canvas tracks and their active state. We'll need `headRotation` and `blendShapes` to store the data we get from MediPipe's computer vision.
+
+```javascript
+// Create the Agora Client
+const client = AgoraRTC.createClient({ 
+  codec: 'vp9',
+  mode: 'live',
+  role: 'host'
+})
+
+const localMedia = {
+  audio: {
+    track: null,
+    isActive: false
+  },
+  video: {
+    track: null,
+    isActive: false
+  },
+  canvas: {
+    track: null,
+    isActive: false
+  },
+}
+
+// Container for the remote streams
+let remoteUsers = {}                
+
+//  store data from facial landmarks
+let headRotation
+let blendShapes
+
+```
+
+### DOMContentLoaded and Event Listeners
+When the page loads, we'll add listeners for the Agora events, the media controls, and the form submission. With the listeners in place, we're ready to show the overlay form.
+
+```javascript
+// Wait for DOM to load
+document.addEventListener('DOMContentLoaded', async () => {
+  // Add the Agora Event Listeners
+  addAgoraEventListeners()
+  // Add listeners to local media buttons
+  addLocalMediaControlListeners()
+  // Get the join channel form & handle form submission
+  const joinform = document.getElementById('join-channel-form')
+  joinform.addEventListener('submit', handleJoin)
+  // Show the overlay form
+  showOverlayForm(true) 
+})
+```
+
+> NOTE: Make sure to add client event listensers before joining the channel, otherwise some events may not get triggered as expected.
+
+## 3D & Avatar Setup
+One of the Prerequisites for this guide is a 3D avatar from ReadyPlaterMe because ReadyPlaterMe provides 3D files that adhere to the name conventions outlined by [Apple's ARKit ARFaceAnchor locations](https://developer.apple.com/documentation/arkit/arfaceanchor/blendshapelocation). These definitions are industry standard and match the output from MediaPipe.
```

Comment:
sp: found 5 instances of ReadyPlayerMe spelled as ReadyPlaterMe

Reply:
fixed